src/components/NavDrawer.jsx

```diff
@@ -86,27 +88,41 @@ export const NavDrawer = () => {
             </ListItemButton>
           </ListItem>
           <Divider />
-          <ListItem disablePadding>
-            <ListItemButton onClick={() => navigate('/cart')}>
-              <ListItemIcon>{<ShoppingCartOutlinedIcon />}</ListItemIcon>
-              <ListItemText primary="Cart" />
-            </ListItemButton>
-          </ListItem>
-          <Divider />
-          <ListItem disablePadding>
-            <ListItemButton onClick={() => navigate('/checkout')}>
-              <ListItemIcon>{<ShoppingCartCheckoutOutlinedIcon />}</ListItemIcon>
-              <ListItemText primary="Checkout" />
-            </ListItemButton>
-          </ListItem>
-          <Divider />
-          <ListItem disablePadding>
-            <ListItemButton onClick={() => navigate('/login')}>
-              <ListItemIcon>{<LoginIcon />}</ListItemIcon>
-              <ListItemText primary="Login" />
-            </ListItemButton>
-          </ListItem>
-          <Divider />
+
+          <div>
+            {!userInfo.userId ? (
+              <ListItem disablePadding>
+                <ListItemButton onClick={() => navigate('/login')}>
+                  <ListItemIcon>{<LoginIcon />}</ListItemIcon>
+                  <ListItemText primary="Login" />
+                </ListItemButton>
+              </ListItem>
+            ) : (
+              <React.Fragment>
+                <ListItem disablePadding>
+                  <ListItemButton onClick={() => navigate('/cart')}>
+                    <ListItemIcon>{<ShoppingCartOutlinedIcon />}</ListItemIcon>
+                    <ListItemText primary="Cart" />
+                  </ListItemButton>
+                </ListItem>
+                <Divider />
+                <ListItem disablePadding>
+                  <ListItemButton onClick={() => navigate('/checkout')}>
+                    <ListItemIcon>{<ShoppingCartCheckoutOutlinedIcon />}</ListItemIcon>
+                    <ListItemText primary="Checkout" />
+                  </ListItemButton>
+                </ListItem>
+                <Divider />
+                <ListItem disablePadding>
+                  <ListItemButton onClick={() => navigate('/login')}>
```

Comment:
While this button is correct, it sends you to the login route, I think this should be doing the same logout method you're using in the navbar